packages/driver/src/cy/commands/sessions.ts

```diff
@@ -0,0 +1,822 @@
+            err = new Error(err)
+          }
+
+          err.stack = $stackUtils.normalizedStack(err)
+
+          err = $errUtils.enhanceStack({
+            err,
+            userInvocationStack: $stackUtils.getUserInvocationStack(err, Cypress.state),
+            projectRoot: Cypress.config('projectRoot'),
+          })
+
+          cy.state('index', index)
+
+          cy.state('onCommandFailed', null)
+
+          _didThrow = err
+
+          return next()
+        })
+
+        const _catchCommand = cy.then(async () => {
+          cy.state('onCommandFailed', null)
+          if (_didThrow) return onFail((_didThrow))
+
+          if (returnVal === false) {
+            // set current command to cy.session for more accurate codeframe
+            cy.state('current', sessionCommand)
+
+            return onFail($errUtils.errByPath('sessions.callback_returned_false', { reason: 'returned false' }))
+            // return onFail((new Error('Your `cy.session` **validate** callback returned false.')))
+          }
+
+          if (returnVal === undefined || Cypress.isCy(returnVal)) {
+            const val = cy.state('current').get('prev')?.attributes?.subject
+
+            if (val === false) {
+              return onFail((new Error('Your `cy.session` **validate** callback resolved false')))
+            }
+          }
+
+          onSuccess()
+        })
+
+        _commandToResume = _catchCommand
+
+        return _catchCommand
+      }
+
+      let hadValidationError = false
+      let onValidationError: Function = (err, log) => {
+        log.set({
+          name: 'Validate Session: invalid',
+          message: '',
+          type: 'system',
+          event: true,
+          state: 'warning',
+        })
+
+        const errorLog = Cypress.log({
+          showError: true,
+          type: 'system',
+          event: true,
+          name: '',
+          message: '',
+        })
+
+        errorLog.error(err)
+        errorLog.set({
+          state: 'warn',
+
+        })
+
+        _log.set({
+          renderProps: () => {
+            return {
+              indicator: 'bad',
+              message: `(recreated) ${_log.get().message}`,
+            }
+          },
+        })
+
+        Cypress.log({ groupEnd: true, emitOnly: true })
+
+        hadValidationError = true
+
+        return runsetup(existingSession)
+        .then(() => {
+          cy.then(() => {
+            return validateSession(existingSession, throwValidationError)
+          })
+          .then(() => {
+            cy.then(async () => {
+              await navigateAboutBlank()
+              Cypress.log({ groupEnd: true, name: '', message: '', emitOnly: true })
+            })
+          })
+        })
+      }
+
+      const throwValidationError = (err) => {
+        err.message += '\n\nThis error occurred in a session validate hook after initializing the session. Because validation failed immediately after session setup we failed the test.'
```

Comment:
This should use [$errUtils.modifyErrMsg](https://github.com/cypress-io/cypress/blob/8550842e3696095c9391e89f30c87af6369ecfb9/packages/driver/src/cypress/error_utils.js#L118) so that it updates the message in the stack as well.

Reply:
85d85eaeeb